### PR TITLE
Improved the API for BiomeMap

### PR DESCRIPTION
**Pull request by WalshyDev**
## Description
BiomeMaps will become a big part of addons supporting 1.18+ and the current implementation doesn't make it easy or fun to implement. So, this PR will expand the API a bit in order to easily support third-party plugins/addons :)

Here is what you'd need to do, what you can do now and an even easier way:
```java
try {
    // What you'd need to do :/
    final BiomeMap<Integer> map = BiomeMap.fromJson(
        new NamespacedKey(this, "test"),
        String.join("", Files.readAllLines(new File(getClass().getResource("/biome-maps/test.json").getFile()).toPath())),
        JsonElement::getAsInt
    );

    // now supported!
    final BiomeMap<Integer> map1 = BiomeMap.fromResource(
        new NamespacedKey(this, "test1"),
        this,
        "/biome-maps/test.json",
        JsonElement::getAsInt
    );

    System.out.println(map1.getOrDefault(Biome.BADLANDS, 0));
    System.out.println(map1.getOrDefault(Biome.BEACH, 0));

    // Simple :)
    final BiomeMap<Integer> map2 = BiomeMap.getIntMapFromResource(
        new NamespacedKey(this, "test2"),
        this,
        "/biome-maps/test.json"
    );
} catch (Exception e) {
    e.printStackTrace();
}
```

## Proposed changes
* Implement a new `BiomeMap.fromResource` method which accepts a JavaPlugin
* Implement some common implementation APIs
  * `getIntMapFromResource(NamespacedKey, JavaPlugin, String path)`
  * `getLongMapFromResource(NamespacedKey, JavaPlugin, String path)`
  * `getStringMapFromResource(NamespacedKey, JavaPlugin, String path)`

## Related Issues (if applicable)
N/A

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/utils/biomes/BiomeMap.java [hunk 1]
```diff
@@ -150,13 +151,34 @@ public String toString() {
 
     @ParametersAreNonnullByDefault
     public static <T> @Nonnull BiomeMap<T> fromResource(NamespacedKey key, String path, BiomeDataConverter<T> valueConverter) throws BiomeMapException {
+        return fromResource(key, Slimefun.instance(), path, valueConverter);
+    }
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/utils/biomes/BiomeMap.java:155]: I completely forgot that different resources will need a different `JavaPlugin` instance. :NotLikeThis:

In my opinion, we can just drop this method and pass `Slimefun.instance()` for our own resources, it'd be too tempting to expose this method here.

Comment (WalshyDev) [reply src/main/java/io/github/thebusybiscuit/slimefun4/utils/biomes/BiomeMap.java]: Agreed

File: src/main/java/io/github/thebusybiscuit/slimefun4/utils/biomes/BiomeMap.java [hunk 2]
```diff
@@ -22,6 +22,7 @@
 import io.github.thebusybiscuit.slimefun4.api.exceptions.BiomeMapException;
 import io.github.thebusybiscuit.slimefun4.api.geo.GEOResource;
 import io.github.thebusybiscuit.slimefun4.implementation.Slimefun;
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/utils/biomes/BiomeMap.java:24]: Import no longer needed
```suggestion
```

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
One unused import otherwise looking good

**Review by TheBusyBiscuit — APPROVED** [review]
LGTM

**Review by Sfiguz7 — APPROVED** [review]

**Pull request merged**